### PR TITLE
Close #138

### LINKED ISSUE #138: better reporting for python 3.3+ chained exceptions
Originally reported by: **Anonymous**

----------------------------------------

With Python 3, py.test doesn't propery display exception chains. The following script
    
    #!python
    def test_chained_exception():
    	try:
    		raise ValueError(17)
    	except Exception as exc:
    		raise ValueError(23) from exc
    
    if __name__ == "__main__":
    	test_chained_exception()
    

when run with Python 3.2 properly outputs the exception chain:

    
    $ python3.2 ~/bug.py
    Traceback (most recent call last):
      File "/Users/walter/bug.py", line 3, in test_chained_exception
        raise ValueError(17)
    ValueError: 17
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "/Users/walter/bug.py", line 8, in <module>
        test_chained_exception()
      File "/Users/walter/bug.py", line 5, in test_chained_exception
        raise ValueError(23) from exc
    ValueError: 23
    

but when run unter py.test only the outermost exception is displayed:

    
    $ python3.2 -mpytest bug.py
    ================ test session starts ================
    platform darwin -- Python 3.2.2 -- pytest-2.1.2
    collected 1 items 
    
    bug.py F
    
    ====================== FAILURES =====================
    _______________ test_chained_exception ______________
    
        def test_chained_exception():
        	try:
        		raise ValueError(17)
        	except Exception as exc:
    >   		raise ValueError(23) from exc
    E     ValueError: 23
    
    bug.py:5: ValueError
    ============= 1 failed in 0.01 seconds ==============
    

----------------------------------------
- Bitbucket: https://bitbucket.org/pytest-dev/pytest/issue/138

### PR DESCRIPTION
**Pull request by roolebo**
The change fixes #138. Exception chain is printed according to [PEP-3134](https://www.python.org/dev/peps/pep-3134/).  
The tracebacks of each exception in the chain are the same as tracebacks of non-chained exceptions in py.test. The tracebacks are delimited by `The above exception was the direct cause of the following exception:` or `During handling of the above exception, another exception occurred:` messages, depending on `__cause__` or `__context__` respectively.  If an exception has both `__cause__` and `__context__`, `__cause__` takes priority. The messages are yellow-colored as it looks better than cyan delimiters:
yellow-exc-sep
cyan-exc-sep

### CODE REVIEW

File: _pytest/runner.py [hunk 1]
```diff
@@ -494,9 +494,12 @@ def importorskip(modname, minversion=None):
     """
     __tracebackhide__ = True
     compile(modname, '', 'eval') # to catch syntaxerrors
+    should_skip = False
     try:
         __import__(modname)
     except ImportError:
+        should_skip = True
+    if should_skip:
```
Comment (The-Compiler) [inline _pytest/runner.py]: This seems like an unrelated change?

Comment (roolebo) [inline _pytest/runner.py]: A chained exception with __context__ set was raised there. Prior to the changes it was ignored, but when I added processing of exception chain it caused a test failure. IIRC the test failed because the trace back of original exception (ImportError) was filtered and no traceback entries were in the entry list.

Comment (RonnyPfannschmidt) [inline _pytest/runner.py]: Please add a note/issue about that, it's a unrelated bug wrt traceback filtering

Comment (roolebo) [inline _pytest/runner.py]: Should I make a separate PR for the issue? Tests won't pass without the fix.

Comment (RonnyPfannschmidt) [inline _pytest/runner.py]: make a issue on the issue tracker and add a link inside a comment of the fix,
then we can sort it out soon

**Comment by nicoddemus** [comment]
I think this change should target the `features` branch...

**Comment by RonnyPfannschmidt** [comment]
oh, right, good call

@roolebo can you please rebase your changes to the features branch
